.github/workflows/docs_deploy.yml

```diff
@@ -19,19 +19,30 @@ jobs:
         with:
           # We need to fetch the whole history so 'reno' can do its job and we can inspect tags.
           fetch-depth: 0
-
       - uses: actions/setup-python@v5
         name: Install Python
         with:
           # Sync with 'documentationPythonVersion' in 'azure-pipelines.yml'.
           python-version: '3.9'
-
       - name: Install dependencies
         run: tools/install_ubuntu_docs_dependencies.sh
 
+      - name: Determine GitHub branch name
+        run: |
+          # PR workflows set the branch they're merging into.
+          if [ -n "$GITHUB_BASE_REF" ]; then
+            BRANCH_NAME="$GITHUB_BASE_REF"
+          # Tags like 1.0.0 and 1.0.0rc1 should point to their stable branch.
+          elif [[ $GITHUB_REF_NAME =~ ^([0-9]+\.[0-9]+) ]]; then
+            BRANCH_NAME="stable/${BASH_REMATCH[1]}"
+          else
+              BRANCH_NAME="$GITHUB_REF_NAME"
```

Comment:
- Minor: is there a reason to use the old-style `[` operator in the first test but the new-style bash `[[` in the second?
- For PRs, do you want to set the base ref to the PR target, or would it be more appropriate to use the commit hash of the temporary PR merge commit?  I'm just thinking that PRs will end up generating incorrect ranges if they change any code, because they'll base the decision about lines of code based on the actual PR, but then output a link to the unchanged `main`.  If you use `GITHUB_SHA` you can probably get an accurate link. 
- What's the reason for having a tag point at the branch it's on and not at the tree of the tag itself?  Seems like the tag would be more reliably accurate.

Reply:
I removed the PR conditional. I realized it's not even relevant to this workflow because it never runs on PRs, which we use Azure for. It was bad bash-ified copy pasta from Runtime. 

Re: PR builds in general, see the PR description for why I think it's fine to at least start with using `main`. If devs do end up actually clicking the link and being confused, then we could add the appropriate logic to Azure in a follow up.

--

Re: tags pointing at branches, this is helpful for us in qiskit/documentation to have a smaller diff when regenerating docs for the same minor release, such as between a tag release vs. `workflow_dispatch`. We try to check the diff when generating docs to make sure there is nothing surprising. Changing every GitHub link is noisy.